### PR TITLE
feat: Add high level type for ROLE command

### PR DESCRIPTION
**Pull request by DCjanus**
resolve https://github.com/redis-rs/redis-rs/issues/1320

### CODE REVIEW

**Comment by nihohit** [comment]
can you please add the `role` function in `commands/mod.rs`, and a test?

**Comment by DCjanus** [comment]
Sorry for delay, I'm trying to imitate other parts of the code, add test cases for my new code, hoping that my imitation is correct and in line with your ideas.

@nihohit

**Review by nihohit — COMMENTED** [review]
Hi, good work, but there's some thing to fix.

File: redis/tests/test_types.rs [hunk 1]
```diff
@@ -79,6 +79,81 @@ mod types {
         }
     }
 
+    #[test]
+    fn test_role_ret() {
```
[the remark this answers is not in the document]
Comment (DCjanus) [reply redis/tests/test_types.rs:83]: > can you add a test to `test_basic` that verifies this works from an actual ROLE call?

I was trying to add a test to `test_basic` for different role of Redis, but looks like it's hard to create an instance as `replica` or `sentinel` from `TestContext`.

Maybe I should add a test case in `test_sentinel`, or import `TestSentinelContext` to `test_basic`, which would you like. @nihohit

[the remark this answers is not in the document]
Comment (nihohit) [reply redis/tests/test_types.rs:83]: you're right. please add it in `test_sentinel`

**Comment by DCjanus** [comment]
Sorry for bother, after a deeper understanding of the existing code, I refactored my code in hopes of making it look more clearer.I hope you wouldn't feel tired because of this.

**Comment by nihohit** [comment]
> I hope you wouldn't feel tired because of this.

No need to apologize. Your contribution is appreciated, as is any extra effort :)

File: redis/tests/test_types.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (DCjanus) [reply redis/tests/test_types.rs:83]: > you're right. please add it in `test_sentinel`

I've add little test code in `test_sentinel`, and make sure it works with `Role` command.

ref: https://github.com/redis-rs/redis-rs/pull/1321/commits/da927cfc3bfb9fbd87cb96fc611431c8b6d3b200

File: redis/src/types.rs [hunk 2]
```diff
@@ -1140,6 +1141,162 @@ impl Deref for InfoDict {
...
+///
+/// [1]: https://redis.io/docs/latest/commands/role/
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct Replica {
+    /// The IP of the replica.
+    pub ip: String,
+    /// The port of the replica.
+    pub port: u16,
+    /// The last acknowledged replication offset.
+    pub offset: i64,
```
Comment (nihohit) [inline redis/src/types.rs:1184]: replication_offset

**Review by nihohit — APPROVED** [review]
Excellent work! 2 tiny fixes, and we're done.

**Comment by DCjanus** [comment]
> Excellent work! 2 tiny fixes, and we're done.

Those changes has been done

**Review by nihohit — APPROVED** [review]

**Comment by nihohit** [comment]
Thank you for the contribution!

**Pull request merged**